libnopegl/src/node_media.c

```diff
@@ -368,6 +373,27 @@ static void media_uninit(struct ngl_node *node)
 #endif
 }
 
+static int filename_changed(struct ngl_node *node)
+{
+    struct media_priv *s = node->priv_data;
+    const int prefetched = s->prefetched;
+
+    if (prefetched)
+        media_release(node);
```

Comment:
Is it really needed ? I think we can just call `media_uninit()`.

Reply:
`media_uninit()` is not going to release the frame because the infrastructure always calls the release callback first, so we have to reproduce that behavior within the node

Reply:
Ok.